internal/clients/gitdiff/gitdiff.go

```diff
@@ -0,0 +1,66 @@
+package gitdiff
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"strings"
+)
+
+const unknownCommitHash = "0000000000000000000000000000000000000000"
+
+// GitDiff client for getting diffs from the command line
+type GitDiff struct {
+	WorkDir    string
+	BaseBranch string
+	BaseSHA    string
+	Head       string
+}
+
+// GetDiff uses the command line to compute the diff
+func (gd *GitDiff) GetDiff() (string, error) {
+	err := os.Chdir(gd.WorkDir)
+	if err != nil {
+		return "", err
+	}
+
+	var diffCmd *exec.Cmd
+	if gd.BaseBranch != "" {
+		err = exec.Command("git", "fetch", "origin", gd.BaseBranch).Run()
+		if err != nil {
+			return "", err
+		}
+		diffCmd = exec.Command("git", "diff", fmt.Sprintf("origin/%s", gd.BaseBranch))
+	} else if gd.BaseSHA == "" || gd.BaseSHA == unknownCommitHash {
+		err = exec.Command("git", "fetch", "origin", gd.Head, "--depth=2").Run()
+		if err != nil {
+			return "", err
+		}
+		diffCmd = exec.Command("git", "diff", "HEAD^", "HEAD")
```

Comment:
This is comparing the current commit with the previous commit. The same output as the previously used `git show format='' ${SHA}`

Reply:
Just checking to make sure this still works for initial commits on new repos?

Reply:
Very good point will need to check that out. I assume this will fail since it is trying to use 2 commits and the branch will only have 1.

Reply:
I have updated this case to use `git show ${SHA} --format=`. `--format=` is the same as `--format=""`, for some reason the command was failing when the `""` was used as the format.